R/get_truth.R

```diff
@@ -221,4 +221,76 @@ preprocess_usafacts <- function (save_location="./data-truth/usafacts/"){
       dplyr::rename(value = inc) %>%
       dplyr::select(date, location, value),
     path = paste0(save_location,"truth_usafacts-Incident Cases.csv"))
+}
+
+
+#' Preprocess raw truth data from JHU CSSE into Cumulative/Incident - Deaths/Cases and write to CSVs
+#' 
+#' @param save_location character specifying the location of to save raw truth data.
+#' Default to "./data-truth/"
+#' 
+#' @export
+preprocess_jhu <- function (save_location="./data-truth/"){
+  # Location name
+  location_names <- covidData::fips_codes[, c("location", "location_name")]
+  
+  # National data
+  national_cases_dframes <- covidData::load_jhu_data(spatial_resolution = "national",
+                                                     temporal_resolution = "daily",
+                                                     measure = "cases")
+  national_deaths_dframes <- covidData::load_jhu_data(spatial_resolution = "national",
+                                                      temporal_resolution = "daily",
+                                                      measure = "deaths")
+
+  # State data
+  state_cases_dframes <- covidData::load_jhu_data(spatial_resolution = "state",
+                                                     temporal_resolution = "daily",
+                                                     measure = "cases")
+  state_deaths_dframes <- covidData::load_jhu_data(spatial_resolution = "state",
+                                                      temporal_resolution = "daily",
+                                                      measure = "deaths")
+  
+  # County data
+  county_cases_dframes <- covidData::load_jhu_data(spatial_resolution = "county",
+                                                     temporal_resolution = "daily",
+                                                     measure = "cases")
+  county_deaths_dframes <- covidData::load_jhu_data(spatial_resolution = "county",
+                                                      temporal_resolution = "daily",
+                                                      measure = "deaths")
+  
+  # Combine to get cases data
+  cases_dframes <- dplyr::bind_rows(national_cases_dframes, state_cases_dframes, county_cases_dframes)
+  cases_dframes <- dplyr::left_join(cases_dframes, location_names, by = "location")
+  cases_dframes$inc[cases_dframes$inc < 0] <- 0
```

Comment:
I'm not sure whether or not we want to do this truncation here. I posted this question on Slack, let's see what the result of that discussion is.

Reply:
We made a decision not to do this truncation.

Reply:
Understood! I have updated this accordingly